lib/identity/stateTransitions/assetLockProof/chain/validateChainAssetLockProofStructureFactory.js

```diff
@@ -0,0 +1,84 @@
+const { Transaction } = require('@dashevo/dashcore-lib');
+const chainAssetLockProofSchema = require('../../../../../schema/identity/stateTransition/assetLockProof/chainAssetLockProof.json');
+
+const convertBuffersToArrays = require('../../../../util/convertBuffersToArrays');
+const InvalidIdentityAssetLockProofCoreHeightError = require('../../../../errors/InvalidIdentityAssetLockProofCoreHeightError');
+const IdentityAssetLockTransactionIsNotFoundError = require('../../../../errors/IdentityAssetLockTransactionIsNotFoundError');
+
+/**
+ * @param {JsonSchemaValidator} jsonSchemaValidator
+ * @param {StateRepository} stateRepository
+ * @param {validateAssetLockTransaction} validateAssetLockTransaction
+ * @returns {validateChainAssetLockProofStructure}
+ */
+function validateChainAssetLockProofStructureFactory(
+  jsonSchemaValidator,
+  stateRepository,
+  validateAssetLockTransaction,
+) {
+  /**
+   * @typedef {validateChainAssetLockProofStructure}
+   * @param {RawChainAssetLockProof} rawAssetLockProof
+   * @returns {ValidationResult}
+   */
+  async function validateChainAssetLockProofStructure(
+    rawAssetLockProof,
+  ) {
+    const result = jsonSchemaValidator.validate(
+      chainAssetLockProofSchema,
+      convertBuffersToArrays(rawAssetLockProof),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    const {
+      coreChainLockedHeight: proofCoreChainLockedHeight,
+      outPoint: outPointBuffer,
+    } = rawAssetLockProof;
+
+    const outPoint = Transaction.parseOutPointBuffer(outPointBuffer);
+    const { outputIndex, transactionHash } = outPoint;
+
+    const rawTransaction = await stateRepository.fetchTransaction(transactionHash);
+
+    if (rawTransaction === null) {
+      result.addError(new IdentityAssetLockTransactionIsNotFoundError(outPointBuffer));
+
+      return result;
+    }
+
+    if (proofCoreChainLockedHeight > rawTransaction.height) {
```

Comment:
```suggestion
    if (proofCoreChainLockedHeight < rawTransaction.height) {
```

The problem if the transaction is mined after we locked the block.